### PR TITLE
test: e2e suite validates an existing PV works after a cluster upgrade

### PR DESCRIPTION
**Pull request by christinalau0**
**Reason for Change**:

Make sure that the upgrade cluster operation does not affect existing PVs.

- The e2e suite validates the CSI Azure disk driver by creating a PV/pod and writing into it
- Set new config variable `CLEAN_PVC=false` in order to keep the PV/pod
- The second time around, the e2e suite will reuse the existing PV/pod
- By default, CLEAN_PVC is true

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: test/e2e/kubernetes/kubernetes_test.go [hunk 1]
```diff
@@ -1794,33 +1794,42 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
...
+					Expect(ready).To(Equal(true))
+				}
 
 				By("Checking that the pod can access volume")
 				valid, err := testPod.ValidatePVC("/mnt/azure", 10, 10*time.Second)
 				Expect(valid).To(BeTrue())
 				Expect(err).NotTo(HaveOccurred())
 
 				// Skip label validation for Azure Disk CSI driver since it currently doesn't apply any label to PV
+				isUsingAzureDiskCSIDriver, _ := eng.HasAddon("azuredisk-csi-driver")
```
Comment (jadarsie) [inline test/e2e/kubernetes/kubernetes_test.go:1832]: define `isUsingAzureDiskCSIDriver` in an outer scope so it is not computed twice

File: test/e2e/kubernetes/kubernetes_test.go [hunk 2]
```diff
@@ -1794,33 +1794,42 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
...
+				By("Checking a persistent volume claim exists")
 				pvcName := "azure-disk" // should be the same as in pvc-azuredisk.yaml
-				pvc, err := persistentvolumeclaims.CreatePersistentVolumeClaimsFromFileWithRetry(filepath.Join(WorkloadDir, "pvc-azuredisk.yaml"), pvcName, "default", 3*time.Second, cfg.Timeout)
-				Expect(err).NotTo(HaveOccurred())
-				// Azure Disk CSI driver in zone-enabled clusters uses 'WaitForFirstConsumer' volume binding mode
-				// thus, pvc won't be available until a pod consumes it
-				isUsingAzureDiskCSIDriver, _ := eng.HasAddon("azuredisk-csi-driver")
-				if !(isUsingAzureDiskCSIDriver && eng.ExpandedDefinition.Properties.HasZonesForAllAgentPools()) {
-					ready, err := pvc.WaitOnReady("default", 5*time.Second, cfg.Timeout)
+				pvc, err := persistentvolumeclaims.Get(pvcName, "default")
```
[the remark this answers is not in the document]
Comment (christinalau0) [reply test/e2e/kubernetes/kubernetes_test.go:1799]: Moved ready check out of block

File: test/e2e/kubernetes/kubernetes_test.go [hunk 3]
```diff
@@ -1794,33 +1794,42 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
...
 
-				By("Launching a pod using the volume claim")
+				By("Checking a pod using the volume claim exists")
 				podName := "pv-pod" // should be the same as in pod-pvc.yaml
-				testPod, err := pod.CreatePodFromFileWithRetry(filepath.Join(WorkloadDir, "pod-pvc.yaml"), podName, "default", 1*time.Second, cfg.Timeout)
-				Expect(err).NotTo(HaveOccurred())
-				ready, err := testPod.WaitOnReady(true, sleepBetweenRetriesWhenWaitingForPodReady, cfg.Timeout)
-				Expect(err).NotTo(HaveOccurred())
-				Expect(ready).To(Equal(true))
+				testPod, err := pod.Get(podName, "default", podLookupRetries)
```
[the remark this answers is not in the document]
Comment (christinalau0) [reply test/e2e/kubernetes/kubernetes_test.go:1816]: err will occur if pod doesn't exist, but moved ready check out of block for pod.Get to address this

File: test/e2e/kubernetes/kubernetes_test.go (hunk 1 above)
Comment (christinalau0) [reply test/e2e/kubernetes/kubernetes_test.go]: fixed

**Review by jadarsie — APPROVED** [review]

**Pull request merged**